### PR TITLE
Add utility Functions

### PR DESCRIPTION
**Pull request by tpfau**
Adding findMetsFromRxns and addGenes utility functions along with the respective tests.

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [] with [X] to check the box)*

### CODE REVIEW

**Comment by tpfau** [comment]
I'll have a look at the efficiency.
Closing for now as addGenes requires the updated automated model modification functions.

**Comment by tpfau** [comment]
Reopening after rebase.
About getCorrespondingRows/Cols:
There is not really a difference between the speed of the current code and getCorrespondingRows/Cols, at least if I directly convert everything to the required boolean vectors for getCorrespondingRows/Cols. If not, the current code is faster.
In addition, if the indexing is based on a logical array it would be necessary to also keep the positional array to properly position the resulting stoichiometries, which creates an unnecessary overhead.

File: src/reconstruction/refinement/addGenes.m [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+%                       used. The following fields will be ignored (as they
+%                       are dependent on the existing model structure):
+%                       rxnGeneMat - The size will be extended, but no associations will be set. 
+% OUTPUTS:
+%
+%    newmodel:     The model structure with the additional reactions.
+%
+% EXAMPLE:
+%
+%    To add genes with specific fields use:
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addGenes.m]: May you use here syntax highlighting?

File: src/reconstruction/refinement/addGenes.m [hunk 2]
```diff
@@ -0,0 +1,59 @@
+function newmodel = addGenes(model,geneIDs,varargin)
+% Add Genes (or a single gene) to the model 
+% USAGE:
```
Comment (laurentheirendt) [inline src/reconstruction/refinement/addGenes.m]: This header needs formatting.

File: src/analysis/exploration/findMetsFromRxns.m [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+%                       will be represented by empty arrays.
+% .. Author: - Thomas Pfau Jan 2018
+
+if ~isnumeric(reactions)
+    rxnInd = findRxnIDs(model, reactions);
+else
+    rxnInd = reactions;
+end
+
+RxnNotInModel = rxnInd==0;
```
Comment (laurentheirendt) [inline src/analysis/exploration/findMetsFromRxns.m]: May you use the `camelToe` convention?

Comment (laurentheirendt) [reply src/analysis/exploration/findMetsFromRxns.m]: May you write `= (rxnInd == 0)` for readability?

File: test/verifiedTests/analysis/testExploration/testFindMetsFromRxns.m [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+reacPos = [4,8];
+
+%Assert both are equal.
+metsFromNames = findMetsFromRxns(model,reacNames);
+metsFromIDs = findMetsFromRxns(model,reacPos);
+assert(isequal(metsFromNames,metsFromIDs));
+
+%Assert that a warning is thrown, but that the output is the same (as all
+% other reactions are ignored)
+invalidReacs = {'A',model.rxns{4},'B',model.rxns{8},'D'};
```
Comment (laurentheirendt) [inline test/verifiedTests/analysis/testExploration/testFindMetsFromRxns.m]: May you rename this as `invalidRxns`?

File: test/verifiedTests/analysis/testExploration/testFindMetsFromRxns.m [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    cstoich = stoich{i};
+    assert(isequal(cstoich(pos),returnedStoich{end-i+1}));
+end
+
+%And now, check that we are fine with invalids.
+
+[returnedMets,returnedStoich] = findMetsFromRxns(model,{'R2','A','B','R1','C'});
+assert(all(cellfun(@(x,y) isempty(setxor(x,y)),mets,returnedMets([4,1]))));
+reacpos = [4,1];
+for i = [1 2]
```
Comment (laurentheirendt) [inline test/verifiedTests/analysis/testExploration/testFindMetsFromRxns.m]: May you write `for i = 1:2` as just before?

**Comment by tpfau** [comment]
The errors are really odd... 
the MILP error on windows is completely random and the linux errors are also quite odd. 
I'll close and reopen to restart the CI on this PR.

**Comment by laurentheirendt** [comment]
Closing to re-trigger CI

**Comment by tpfau** [comment]
retriggering ci again

**Comment by tpfau** [comment]
Anything else here?

**Pull request merged**